CribblyBackend.UnitTests/ControllerTests/TeamControllerTests.cs

```diff
@@ -69,6 +69,41 @@ public async Task GetById_ShouldReturnTeamAndOkStatus()
             Assert.Equal(expTeam.Name, actTeam.Name);
             Assert.Equal(expTeam.Players, actTeam.Players);
         }
+        [Fact]
+        public async Task GetAll_ShouldReturnTeamsAndOkStatus()
+        {
+            var expTeams = new List<Team>();
+            for (int i = 1; i <= 30; i++)
+            {
+                Team team = new Team()
+                {
+                    Id = i,
+                    Name = $"test Team {i}",
+                    Players = new List<Player>()
+                };
+                for (int j = 1; j <= 2; j++)
+                {
+                    Player expPlayer = new Player()
+                    {
+                        Id = i,
+                        Email = $"test{i}@test.com",
+                        Name = $"test player {i}"
+                    };
+                    team.Players.Add(expPlayer);
+                }
+            }
+
+
+            mockTeamService.Setup(x => x.GetAll()).ReturnsAsync(expTeams);
+            var result = await TeamController.GetAll();
+            var okResult = Assert.IsType<OkObjectResult>(result);
+            var actTeams = Assert.IsType<List<Team>>(okResult.Value);
+            Assert.Equal(expTeams.Count, actTeams.Count);
+            foreach (Team team in actTeams)
+            {
+                Assert.Equal(2, team.Players.Count);
```

Comment:
:bulb: If you want to be more specific about asserting a collection, you can use `Assert.Collection`; not sure if that's necessary here, but something to keep in mind

Reply:
Stack Overflow says I can go ```Assert.Equal(expTeams, actTeams);```. I am good with that if you are

Reply:
Does that do a deep equality comparison on each object in the list?

Reply:
It requires matches on length and order - is that deep equality?

Reply:
Sounds aight to me :man_shrugging: